apps/web/modules/ee/audit-logs/lib/utils.ts

```diff
@@ -0,0 +1,348 @@
+    } else if (oldObject) {
+      changes = redactPII(oldObject);
+    }
+
+    const auditEvent: TAuditLogEvent = {
+      actor: { id: userId, type: userType },
+      action: actionType,
+      target: { id: targetId, type: targetType },
+      timestamp: new Date().toISOString(),
+      organizationId,
+      status,
+      ipAddress: AUDIT_LOG_GET_USER_IP ? ipAddress : UNKNOWN_DATA,
+      apiUrl,
+      ...(changes ? { changes } : {}),
+    };
+
+    if (status === "failure") {
+      auditEvent.eventId = eventId;
+    }
+
+    await logAuditEvent(auditEvent);
+  } catch (logError) {
+    logger.error(logError, "Failed to create audit log event");
+  }
+}
+
+/**
+ * Logs an audit event.
+ * The audit logging runs in the background to avoid blocking the main request.
+ *
+ * @param actionType - The type of action to audit.
+ * @param targetType - The type of target (e.g., "segment", "survey").
+ * @param userId - The ID of the user performing the action.
+ * @param targetId - The ID of the target (e.g., segment ID, survey ID).
+ * @param organizationId - The ID of the organization.
+ * @param ipAddress - The IP address of the user.
+ * @param oldObject - The old object (optional).
+ * @param newObject - The new object (optional).
+ * @param status - The status of the action ("success" or "failure").
+ * @param apiUrl - The URL of the API request.
+ **/
+export async function queueAuditEventBackground({
+  actionType,
+  targetType,
+  userId,
+  userType,
+  targetId,
+  organizationId,
+  oldObject,
+  newObject,
+  status,
+  eventId,
+  apiUrl,
+}: {
+  actionType: TAuditActionType;
+  targetType: TAuditTarget;
+  userId: string;
+  userType: TActor;
+  targetId: string | undefined;
+  organizationId: string;
+  oldObject?: any;
+  newObject?: any;
+  status: TAuditStatus;
+  eventId?: string;
+  apiUrl?: string;
+}) {
+  setImmediate(async () => {
+    const ipAddress = await getClientIpFromHeaders();
+
+    await buildAndLogAuditEvent({
+      actionType,
+      targetType,
+      userId,
+      userType,
+      targetId,
+      organizationId,
+      ipAddress,
+      status,
+      oldObject,
+      newObject,
+      eventId,
+      apiUrl,
+    });
+  });
+}
+
+/**
+ * Logs an audit event.
+ * This function will block the main request. Use it only in edge runtime functions, like api routes.
+ *
+ * @param actionType - The type of action to audit.
+ * @param targetType - The type of target (e.g., "segment", "survey").
+ * @param userId - The ID of the user performing the action.
+ * @param targetId - The ID of the target (e.g., segment ID, survey ID).
+ * @param organizationId - The ID of the organization.
+ * @param ipAddress - The IP address of the user.
+ * @param oldObject - The old object (optional).
+ * @param newObject - The new object (optional).
+ * @param status - The status of the action ("success" or "failure").
+ * @param apiUrl - The URL of the API request.
+ **/
+export async function queueAuditEvent({
```

Comment:
In most of our code we use the variable assignment for functions instead of the `function` syntax, e.g.
```
export const queueAuditEvent = (....) {};
```

please use the same syntax here for consistency.